scrape/scrape.go

```diff
@@ -359,7 +396,8 @@ func (sp *scrapePool) Sync(tgs []*targetgroup.Group) {
 
 	var all []*Target
 	sp.mtx.Lock()
-	sp.droppedTargets = []*Target{}
+	defer sp.mtx.Unlock()
```

Comment:
This moves the unlock to after the sync, is that intentional?

Reply:
Curious about this as well, is the moving of the lock required for any added functionality that I am missing. Would be good to comment why this needs to be here if thats the case.

Reply:
So I think it makes no sense do drop the lock here if it is to just take it again in the sync() method. this is the only place where sync() is called. I can change it back if you really want but I'd like to keep this as is.

Reply:
The defer should be right after the lock.